cocos/renderer/CCBatchCommand.cpp

```diff
@@ -69,11 +97,46 @@ void BatchCommand::execute()
     // Set material
     _shader->use();
     _shader->setUniformsForBuiltins(_mv);
-    GL::bindTexture2D(_textureID);
-    GL::blendFunc(_blendType.src, _blendType.dst);
 
     // Draw
-    _textureAtlas->drawQuads();
+    if (_textureAtlas)
+    {
+        GL::bindTexture2D(_textureID);
+        GL::blendFunc(_blendType.src, _blendType.dst);
+        _textureAtlas->drawQuads();
+    }
+    else if (_batch)
+    {
+        if (_textureID)
+            GL::bindTexture2D(_textureID);
+        GL::blendFunc(_blendType.src, _blendType.dst);
+        auto drawn = _batch->draw(_start, _count);
+        CC_INCREMENT_GL_DRAWN_BATCHES_AND_VERTICES(1, drawn);
+    }
+    else
+    {
+        CCASSERT(false, "BatchCommand::execute - nothing to draw");
+    }
+}
+
+unsigned  BatchCommand::getStart() const
+{
+    return _start;
+}
+
+unsigned  BatchCommand::getCount() const
+{
+    return _count;
+}
+
+void BatchCommand::setStart(unsigned start)
```

Comment:
yes it always picks the optimal type. It will be whatever the native register size is on the cpu. It's part of the C standard, so it's guaranteed cross compiler.